YOLayout/YOLayout.m

```diff
@@ -65,7 +65,17 @@ - (CGSize)sizeThatFits:(CGSize)size {
 }
 
 - (CGRect)setFrame:(CGRect)frame view:(id)view sizeToFit:(BOOL)sizeToFit {
-  return [self setFrame:frame view:view options:(sizeToFit ? YOLayoutOptionsSizeToFit : 0)];
+
+  YOLayoutOptions options = 0;
+  // If UILabel and 1 line, we need to fix size to fit options to constrain width,
+  // since UILabel will return a size > the width we specify.
+  if (sizeToFit && [view isKindOfClass:UILabel.class] && ((UILabel *)view).numberOfLines == 1) {
```

Comment:
yeah ok.. part of me thinks the "Vertical" should be implied but hey its
objective-c, lets be verbose

On Mon, Jan 12, 2015 at 5:03 PM, John Boiles <notifications@github.com>
wrote:

> In YOLayout/YOLayout.m
> <https://github.com/YOLayout/YOLayout/pull/10#discussion-diff-22836734>:
>
> > @@ -65,7 +65,17 @@ - (CGSize)sizeThatFits:(CGSize)size {
> >  }
> >
> >  - (CGRect)setFrame:(CGRect)frame view:(id)view sizeToFit:(BOOL)sizeToFit {
> > -  return [self setFrame:frame view:view options:(sizeToFit ? YOLayoutOptionsSizeToFit : 0)];
> > +
> > +  YOLayoutOptions options = 0;
> > +  // If UILabel and 1 line, we need to fix size to fit options to constrain width,
> > +  // since UILabel will return a size > the width we specify.
> > +  if (sizeToFit && [view isKindOfClass:UILabel.class] && ((UILabel *)view).numberOfLines == 1) {
>
> What do you think about naming this: sizeToFitVerticalInFrame:view:
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/YOLayout/YOLayout/pull/10/files#r22836734>.
>